### ISSUE TITLE
componentDidMount of current slide

### ISSUE THREAD

**Issue by RaidoS** [express-labs/pure-react-carousel#51]
Are there any tricks to get `componentDidMount` of current slide. Suppose, i want change behavior of slide content ?

**Comment by mrbinky3000** [comment]
It's possible for a slide to be mounted first before a current slide is chosen.  For example, you have three slides.  All three slides mount.  Then you click on a dot to change to the second slide.   The second slide is now current, but it was mounted quite some time ago.

It might be better to wrap your slide's child content with a custom component that uses the new WithStore() higher order component.


```html
<!--- pseudocode -->
<Slider>
  <Slide>
    <MyCustomComponent>
      [ your slide content here ]
    </MyCustomComponent>
  </Slide>
</Slider>
```

You can use WithStore to pass in the current slide (currentSlide), the total number of slides (totalSlides), and the number of visible slides in your slider (visibleSlides)

With those three values, you can determine which slides are currently visible in your slider and your custom component can alter stuff as you see fit.

You can even make your own higher order component that returns a custom version of `<Slide />`

Basically, unless I misunderstood your question, there are several ways to tackle this.    Closing for now, but will reopen if I didn't understand your question.
